app/components/download-spreadsheet.js

```diff
@@ -1,14 +1,22 @@
 import Component from '@ember/component';
 import { inject as service } from '@ember/service';
+import { getProperties } from '@ember/object';
 
 export default Component.extend({
   tagName: '',
   data: null, // []
-  mode: '',
-  tab: '',
+  // excludedProperties is an array of keys that are excluded from each condition (current mode Census, change mode Census, current mode ACS, change mode ACS)
+  excludedProperties: ['codingThresholds', 'rowConfig', 'notinprofile', 'variablename', 'year', 'is_most_recent', 'geotype', 'producttype', 'release_year', 'unittype'],
+  // create arrays of keys that should be included in each of the four conditions
+  censusCurrent: ['numGeoids', 'profile', 'category', 'variable', 'dataset', 'estimate', 'percent', 'comparison_estimate', 'comparison_percent', 'difference_estimate', 'difference_percent'],
+  censusChange: ['numGeoids', 'profile', 'category', 'variable', 'dataset', 'previous2000_estimate', 'previous2000_percent', 'estimate', 'percent', 'change_estimate', 'change_percent', 'change_percentage_point'],
+  acsCurrent: ['numGeoids', 'profile', 'category', 'variable', 'dataset', 'base', 'estimate', 'moe', 'cv', 'percent', 'percent_moe', 'is_reliable', 'comparison_estimate', 'comparison_moe', 'comparison_cv', 'comparison_percent', 'comparison_percent_moe', 'comparison_is_reliable', 'difference_estimate', 'difference_moe', 'difference_reliable', 'difference_percent', 'difference_percent_moe', 'difference_percent_reliable'],
+  acsChange: ['numGeoids', 'profile', 'category', 'variable', 'dataset', 'base', 'previous0610_estimate', 'previous0610_moe', 'previous0610_cv', 'previous0610_percent', 'previous0610_percent_moe', 'estimate', 'moe', 'cv', 'percent', 'percent_moe', 'is_reliable', 'change_estimate', 'change_moe', 'change_reliable', 'change_percent', 'change_percent_moe', 'change_percent_reliable', 'change_percentage_point', 'change_percentage_point_moe', 'change_percentage_point_reliable'],
```

Comment:
Non-blocking nit/thought:

These are static pieces of information. I usually like to make these as exportable constants above the class (line 4). There are two reasons for this:

1. Static information is exportable from here & importable from somewhere else, making it easy to reuse the information. 
2. Defining them on the class makes them part of the instance, and they don't really depend on any state. For example, they aren't being changed by some piece of information being passed into the component. 

I think this approach helps make things clearer, too, since the stateful part of the code is often the most important part, and you keep static things out of the way.